### PR TITLE
[doc] Fix typo and formatting in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Architector4**
# Objective

Issue: There is a typo in `CONTRIBUTING.md` ("then" used in place of "them"). There is also an inconsistency of usage of periods at ends of items in lists, and one section is written with non-breaking spaces without good reason.

## Solution

Fix the aforementioned typo and consistency issues.

### CODE REVIEW

**Comment by Architector4** [comment]
Whoops, forgot to mention making items of the first list in the document start with an uppercase letter, and removal of one probably insignificant newline.

Usage of newlines seems sporadic in the document in general, but I felt it would make this a bit more annoying of a commit to deal with, and so decided to not do anything with that lol

**Review by mockersf — APPROVED** [review]

**Review by tjhorner — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -78,17 +78,17 @@ Check out our dedicated [Bevy Organization document](/docs/the_bevy_organization
...
-* D: Difficulty (e.g. D-Complex, D-Good-First-Issue)
-* O: Operating System (e.g. O-Linux, O-Web, O-Windows)
-* P: Priority (e.g. P-Critical, P-High)
-* S: Status (e.g. S-Blocked, S-Controversial, S-Needs-Design)
+* A: Area (e.g. A-Animation, A-ECS, A-Rendering).
+* C: Category (e.g. C-Breaking-Change, C-Code-Quality, C-Docs).
+* D: Difficulty (e.g. D-Complex, D-Good-First-Issue).
+* O: Operating System (e.g. O-Linux, O-Web, O-Windows).
+* P: Priority (e.g. P-Critical, P-High).
+* S: Status (e.g. S-Blocked, S-Controversial, S-Needs-Design).
```
Comment (Jondolf) [inline CONTRIBUTING.md:91]: I wouldn't use periods here. Bulleted list items should generally use periods if they either contain a full sentence or complete the introductory stem sentence (the sentence before ":"), which is not the case here. Stylistically, ")." also doesn't look as clean in my opinion.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -78,17 +78,17 @@ Check out our dedicated [Bevy Organization document](/docs/the_bevy_organization
 
 Our merge strategy relies on the classification of PRs on two axes:
 
-* How controversial are the design decisions
-* How complex is the implementation
+* How controversial are the design decisions.
+* How complex is the implementation.
```
Comment (Jondolf) [inline CONTRIBUTING.md:82]: These are questions, so if we want punctuation here, the sentences should end with a question mark.

**Review by Jondolf — COMMENTED** [review]
Some (maybe subjective) punctuation nits, otherwise looks good to me :)

**Review by doonv — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (rparrett) [reply CONTRIBUTING.md:82]: Or rephrase as e.g. `How complex the implementation is.`

**Comment by Architector4** [comment]
Oop-

I'd have fixed the above too, but was asleep. Ah well, at least I hope I made somewhat of an improvement lol

**Pull request merged**